src/main/java/io/bittiger/ads/activity/TopKAds.java

```diff
@@ -0,0 +1,51 @@
+package io.bittiger.ads.activity;
+
+import io.bittiger.ads.model.Ad;
+
+import java.util.*;
+
+public class TopKAds {
+    private static TopKAds instance = null;
+    protected TopKAds() {
+
+    }
+    public static TopKAds getInstance() {
+        if (instance == null) {
+            instance = new TopKAds();
+        }
+        return instance;
+    }
+    public List<Ad> selectTopKAds(List<Ad> rankedAds, int k) {
+        LinkedList<Ad> selectedAds = new LinkedList<Ad>();
+        if (rankedAds == null || rankedAds.size() == 0) {
+            return selectedAds;
+        }
+        PriorityQueue<Ad> minHeap = new PriorityQueue<Ad>(k, new Comparator<Ad>() {
+            public int compare(Ad o1, Ad o2) {
+                if (o1.getRankScore() == o2.getRankScore()) {
+                    return 0;
+                }
+                return o1.getRankScore() > o2.getRankScore() ? -1 : 1;
```

Comment:
I guess you implemented the **max** heap not the **min** one. Can you take a look at this part? @sleephu